structure-macro-impl/src/lib.rs

```diff
@@ -1,65 +1,71 @@
 #![recursion_limit = "128"]
 
-#[macro_use]
-extern crate proc_macro_hack;
-#[macro_use]
-extern crate quote;
+extern crate proc_macro;
+extern crate alloc;
 
-use std::mem;
-use std::os::raw::c_void;
-use std::string::String;
-use quote::{Tokens, Ident};
+use alloc::string::String;
```

Comment:
In the proc-macro you can still use `std`. No need for `alloc`.

Reply:
Yes, but would that work in `no_std`? My understanding is that the proc-macro would break under `no_std` without `alloc`. Could be wrong.

Reply:
The proc-macro code is compiled for the host, since it runs on the host at compile time. So no-std isn't needed / doesn't make sense for the proc-macro crate.

Reply:
Resolved in e88f7d1.